src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -587,6 +588,83 @@ int sd_dhcp6_lease_get_captive_portal(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_get_vendor_option(sd_dhcp6_lease *lease) {
+        int r;
+        sd_dhcp6_option **options = NULL;
+        size_t n_vendor_options;
+
+        assert(lease);
+
+        r = set_dump_sorted(lease->vendor_options, (void***)&options, &n_vendor_options);
```

Comment:
- This may be called several times for the same lease. Let's cache the sorted result to the lease.
- Please make the function take an argument to store options, otherwise this leaks memory.
- Please export the function, that is, prefix with `sd_` and declare it in src/systemd/sd-dhcp6-lease.h.
- `get_vendor_option()` -> `get_vendor_options()`
```c
int sd_dhcp6_lease_get_vendor_options(sd_dhcp6_lease *lease, sd_dhcp6_option ***ret) {
        int r;

        assert_return(lease, -EINVAL);

        if (set_isempty(lease->vendor_options))
                return -ENODATA;

        if (ret) {
                if (!lease->sorted_vendor_options) {
                        r = set_dump_sorted(lease->vendor_options, (void***) &lease->sorted_vendor_options, NULL);
                        if (r < 0)
                                return r;
                }

                *ret = lease->sorted_vendor_options;
        }

        return set_size(lease->vendor_options);
}
```

Reply:
Fixed it.